Src/RE2.Managed/Regex2.cs

```diff
@@ -269,13 +272,33 @@ public static void ClearRegexes()
                         submatchStrings.Add(submatchString);
                     }
 
+                    // Build MatchGroup list
+                    matchGroups = new List<MatchGroup>();
```

Comment:
>List [](http://example.com/codeflow?start=38&length=4)

change to dictionary<string, matchgroup>
for the annonymous, use 0->n
for the named, use groupname

this is the same way .NET Regex works

Reply:
that would gives us O(1) access to the named groups

Reply:
Please review updated interface.